src/muxer.ts

```diff
@@ -89,12 +96,25 @@ export class DataChannelMuxer implements StreamMuxer {
             open: 0
           }
         },
-        closeCb: init?.onStreamEnd
+        closeCb: this.wrapStreamEnd(init?.onIncomingStream)
       })
 
       if ((init?.onIncomingStream) != null) {
         init.onIncomingStream(stream)
       }
+      this.streams.push(stream)
+    }
+
+    if (init?.onIncomingStream != null) {
+      streams.forEach((_stream) => {
+        if (_stream.stat.timeline.close != null) {
+          return
+        }
+        (_stream as WebRTCStream).closeCb = this.wrapStreamEnd(init.onStreamEnd)
+        this.streams.push(_stream)
+        // @ts-expect-error
+        void Promise.resolve().then(() => init?.onIncomingStream(_stream))
+      })
     }
```

Comment:
`s.stat.timeline.close != null` would filter out all streams that are open, so that condition needs to be inverted. Other than that, I'm testing this change.